diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -183,28 +189,36 @@ controller_interface::return_type DiffDriveController::update(
   }
   else
   {
-    double left_position_mean = 0.0;
-    double right_position_mean = 0.0;
+    double left_feedback_mean = 0.0;
+    double right_feedback_mean = 0.0;
     for (size_t index = 0; index < wheels.wheels_per_side; ++index)
     {
-      const double left_position = registered_left_wheel_handles_[index].position.get().get_value();
-      const double right_position =
-        registered_right_wheel_handles_[index].position.get().get_value();
+      const double left_feedback = registered_left_wheel_handles_[index].feedback.get().get_value();
+      const double right_feedback =
+        registered_right_wheel_handles_[index].feedback.get().get_value();
 
-      if (std::isnan(left_position) || std::isnan(right_position))
+      if (std::isnan(left_feedback) || std::isnan(right_feedback))
       {
         RCLCPP_ERROR(
-          logger, "Either the left or right wheel position is invalid for index [%zu]", index);
+          logger, "Either the left or right wheel %s is invalid for index [%zu]", feedback_type(),
+          index);
         return controller_interface::return_type::ERROR;
       }
 
-      left_position_mean += left_position;
-      right_position_mean += right_position;
+      left_feedback_mean += left_feedback;
+      right_feedback_mean += right_feedback;
     }
-    left_position_mean /= wheels.wheels_per_side;
-    right_position_mean /= wheels.wheels_per_side;
+    left_feedback_mean /= wheels.wheels_per_side;
+    right_feedback_mean /= wheels.wheels_per_side;
 
-    odometry_.update(left_position_mean, right_position_mean, current_time);
+    if (odom_params_.position_feedback)
+    {
+      odometry_.update(left_feedback_mean, right_feedback_mean, current_time);
+    }
+    else
+    {
+      odometry_.updateFromVelocity(left_feedback_mean, right_feedback_mean, current_time);
```

Comment:
Shouldnt this be `left_feedback_mean * dt` and `right_feedback_mean * dt` since `left_vel` and `right_vel` used in `integrateExact` are supposed to be the position deltas and not the instantaneous velocities? In the `position_feedback == true` path, `left_vel` and `right_vel` [are passed as position deltas](https://github.com/ros-controls/ros2_controllers/blob/dedb120993c6a95e240c2cf7d7730767ab701296/diff_drive_controller/src/odometry.cpp#L62). The fork that the feedback_param was based off of also [multiplies by dt](https://github.com/clearpathrobotics/ros_controllers/blob/22e9c2cf64b029abd88eafedae6b988a3845585f/diff_drive_controller/src/odometry.cpp#L128)

Reply:
Sorry, I'm from mobile now, but your line of thought seems reasonable at a first look. In these days I'm on vacation so I can't evaluate and fix, maybe it is worth pushing a new PR with the fixes you have in mind explaining there what's wrong with current code.

Reply:
It also doesn't really make sense to me why both pose and twist get computed from vel when this param is enabled. Doesn't it make more sense to compute odom pose using position deltas calculated from the position interface and then use instantaneous vel from the velocity interface to calculate the odom twist? This is the approach that the clearpath fork ended up using, allowing for two params to disable this behavior (originally implemented in https://github.com/clearpathrobotics/ros_controllers/pull/18, but available in their latest indigo branch). Thoughts on this?

Reply:
@wmmc88 do you opened any issue or PR related to this?

For your last message, I undestand that you are talking about using simultaneously the feedback position and velocity interfaces. Is is correct? In my undestanding, for now there is only support here for one of the two IF/s at time, but using both if available could be interesting (not to be seen as a bug, but as a feature enhancement).

Reply:
> @wmmc88 do you opened any issue or PR related to this?

nope. havent opened an issue and haven't had any time to address it myself in a pr.

> For your last message, I undestand that you are talking about using simultaneously the feedback position and velocity interfaces. Is is correct? In my undestanding, for now there is only support here for one of the two IF/s at time, but using both if available could be interesting (not to be seen as a bug, but as a feature enhancement).

yeah this wasn't meant to be an issue with this pr. I'm more just suggesting as an enhancement that the pose be calculated by the position interface data and the twist be calculated from the velocity interface data. I believe that is a more optimal solution and that is what the linked clearpath fork did(they reverted a change similar to this pr and went with the pose from position/twist from velocity approach).

Reply:
I reviewed now, and you are true. Since the code calls "vel" the position delta (misleadingly), I made confusion. A multiplication with `dt` will convert back instantaneous velocity to deltas. Might now also consider to do the multiplication inside updateFromVelocity() to hide this detail at interface level. Will open PR now.